core/google/cloud/_helpers.py

```diff
@@ -309,20 +179,7 @@ def _determine_default_project(project=None):
     :returns: Default project if it can be determined.
     """
     if project is None:
-        project = _get_production_project()
-
-    if project is None:
-        project = _file_project_id()
-
-    if project is None:
-        project = _default_service_project_id()
-
-    if project is None:
-        project = _app_engine_id()
-
-    if project is None:
-        project = _compute_engine_id()
-
+        _, project = google.auth.default()
```

Comment:
Definitely need a refactor to avoid dropping the credentials on the floor, but fine for now.